client/src/components/Register.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+
+export default function Register(props) {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const handleRegister = ev => {
+    ev.preventDefault();
+    if (!username || !password)
+      return props.setError('Please enter username and password.');
+    const credentials = { username, password };
+    setUsername('');
+    setPassword('');
+    axios
+      .post('http://localhost:3300/api/register', credentials)
+      .then(res => {
+        props.setError(null);
+        localStorage.setItem('jwt', res.data.token);
+        props.history.push('/jokes');
+      })
+      .catch(err =>
+        props.setError(
+          err.response
+            ? err.response.data.error
+            : 'There was an error while attempting registration.'
+        )
+      );
+  };
+  return (
+    <form className="register auth" onSubmit={handleRegister}>
+      <input
+        type="text"
+        placeholder="Username"
+        onChange={ev => setUsername(ev.target.value)}
+        value={username}
+      />
+      <input
+        type="password"
+        placeholder="Password"
+        onChange={ev => setPassword(ev.target.value)}
+        value={password}
+      />
+      <button>Register</button>
+    </form>
```

Comment:
It looks like this component (`Register`) and `Login` are super similar. Do you think you could combine them into one? The parts you'd need to account for being different are the class names (partially), the submit handlers, the error messages, and the text for the form button. It might seem a little tedious at the moment, but it's definitely good to have some practice with this so that you can use it where it's certainly nicer to have.